pkg/scanners/scannerv4/scannerv4.go

```diff
@@ -202,3 +208,92 @@ func (s *scannerv4) GetVulnerabilities(image *storage.Image, components *types.S
 
 	return imageScan(image.GetMetadata(), vr), nil
 }
+
+func (s *scannerv4) GetNodeVulnerabilityReport(node *storage.Node, indexReport *v4.IndexReport) (*v4.VulnerabilityReport, error) {
+	nodedigest, err := name.NewDigest("registry/repository@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
+	if err != nil {
+		log.Errorf("Failed to parse digest from node %q: %v", node.GetName(), err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), scanTimeout)
+	defer cancel()
+
+	vr, err := s.scannerClient.GetVulnerabilities(ctx, nodedigest, indexReport.GetContents())
+	if err != nil {
+		log.Errorf("Failed to create vulnerability report: %v", err)
+		return nil, errors.Wrap(err, "Failed to create vulnerability report")
+	}
+
+	return vr, nil
+}
+
+func (s *scannerv4) GetNodeInventoryScan(node *storage.Node, inv *storage.NodeInventory, ir *v4.IndexReport) (*storage.NodeScan, error) {
+	if ir == nil && inv != nil {
+		return nil, errors.New("Received Scanner v2 data for Scanner v4. Exiting.")
+	}
+	vr, err := s.GetNodeVulnerabilityReport(node, ir)
+	if err != nil {
+		log.Errorf("Failed to create vulnerability report: %v", err)
+		return nil, errors.Wrap(err, "Failed to create vulnerability report")
+	}
+	log.Infof("Received Vulnerability Report with %d vulnerabilities: %v", len(vr.Vulnerabilities), vr)
```

Comment:
I think, I mentioned that I would like to see a log.Info when the scanning succeeds - and here it is. However, I think it better fits into the pipeline code in Central. Will not insist if you decide to keep it here, but then the `vr` may be huge and some more compact information may be preferred (even for the short time this log will live before you implement the follow-ups).

Reply:
I lean towards keeping the log here, because this is the only place we can access the "raw" VulnerabilityReport before conversion. But I've improved the log message :)